### ISSUE TITLE
Performing "update all" does not give any user feedback

### ISSUE THREAD

**Issue by papernoise** [monome/maiden#183]
When clicking the "update all" button in the project manager, the UI stays on the "please wait…" window. Scripts are being updated it seems, but maiden does not give any feedback about what's going on and about having successfully completed or not the update.

When clicking that button again, I get the "failed" message and the list of scripts that were not updated because they are already up to date.

**How to reproduce:**
Click on the "update all button" then acknowldge hat modifications will be lost.
The window will say "please wait… " and – if there is scripts that need updating – nothing will happen after that. If not you'll get the "failed" window.
You can click the checkmark to close the window

then click on "update all" again and you will get the "failed" window with the list of the scripts that it has tried to update.

**OS**: MacOS Catalina/Sierra
**Browser**: Firefox 76.0.1 (64bit)
**Extentions that could interfere**: none that I know. I've tried to disable them all and the problem persists.

**Comment by ngwese** [comment]
Agreed, the current UI is far from ideal. The two primary bodies of work which need to be addressed are:

- The `maiden` HTTP backend does not currently provide any progress information which the update requests are being performed
- The `maiden` frontend doesn't currently use a third party ui component library so a spinner or progress bar ui needs to be implemented or a third party library needs to be identified (and the component styled to match)

**Comment by ngwese** [comment]
The #202 hopefully addresses some, but not all,  of the concerns noted above - in particular there is still no progress feedback during long running operations such as "update all"

**Comment by ngwese** [comment]
This is still an issue which people run into frequently.

One strategy would improve the current implementation without having to change anything in the backend would be:
- Obtain the list of installed projects
- Trigger updates serially instead of all in parallel/asynchronously
- As updates succeed/fail push a progress bar forward

The currently implement kicks off update requests for all installed scripts in parallel IIRC. This is problematic when people have a large number of scripts installed and/or latency/bandwidth to the script location is high. In theory it is possible with the current implementation for updates to a large number of scripts to essentially cause the maiden backend to overwhelm the norns device and/or run out of cpu/memory.
